frontend/tests/controllers/ContestCreateTest.php

```diff
@@ -150,6 +151,42 @@ public function testCreateContestWithInvalidAlias() {
             $this->assertEquals('parameterInvalid', $e->getMessage());
         }
     }
+    /**
+     * A PHPUnit data provider for all the plagiarism threshold values in a
+     * contest.
+     *
+     * @return list<array{0:bool, 1:int}>
+     */
+    public function plagiarismThresholdProvider(): array {
+        return [
+            [true, 90],
+            [false, 0],
+        ];
+    }
+
+    /**
+     * Check if the plagiarism value is stored correctly in the database.
+     * @dataProvider plagiarismThresholdProvider
+     */
+    public function testToValidatePlagiarismThresholdValue(
+        bool $checkPlagiarism
+        int $plagiarismThresholdExpected
+    ) {
+        // Create a contest
+        $contestData = \OmegaUp\Test\Factories\Contest::createContest(
+            new \OmegaUp\Test\Factories\ContestParams([
+                'plagiarism_threshold' => $checkPlagiarism,
+            ]);
+        )
+        $checkPlagiarismDuplicate = \OmegaUp\DAO\Contests::getByAlias(
+            $contestData['request']['alias']
+        );
+
+        $this->assertEquals(
+            $checkPlagiarismDuplicate->plagiarism_threshold,
+            $plagiarismThresholdExpected
+        );
```

Comment:
It seems like `assertEquals` function is evaluating the expressions as boolean, The way to fix that is using [`assertSame` ](https://phpunit.de/manual/6.5/en/appendixes.assertions.html#appendixes.assertions.assertSame) function. It evaluates the type and the value:
```suggestion
        $this->assertSame(
            $checkPlagiarismDuplicate->plagiarism_threshold,
            $plagiarismThresholdExpected
        );
```